kyo-core/shared/src/main/scala/kyo/Fiber.scala

```diff
@@ -248,7 +216,7 @@ object Fiber extends FiberPlatformSpecific:
           * @return
           *   A new Fiber that runs with interrupt masking
           */
-        def mask(using Frame): Fiber[E, A] < IO = IO(self.mask)
+        def mask(using Frame): Fiber[E, A] < IO = IO.Unsafe(Unsafe.mask(self)())
```

Comment:
Explicitly call the `mask` extension method via its companion object to avoid the compiler inferring the incorrect version.